### DISCUSSION TITLE
Airflow task run logs did not write to stdout by enabling AIRFLOW__ELASTICSEARCH__WRITE_STDOUT

### DISCUSSION THREAD

**Discussion by kangtao** [apache/airflow#18367] [General]
Hi,

I deploy airflow 1.10.14 to Azure Kubernetes Service and I want airflow to write task logs to stdout so k8s logs can capture the task logs. Below are my configs. I do not really have elasticsearch, but I just want to make sure that airflow not only stores the task logs to local dag task folders, it also writes the log to stdout. I can see dag_processor_manager log, scheduler logs from k8s logs, but not task run logs. Is this expected?

"AIRFLOW__ELASTICSEARCH__WRITE_STDOUT": "True",
"AIRFLOW__ELASTICSEARCH__JSON_FORMAT": "True",
"AIRFLOW__ELASTICSEARCH__JSON_FIELDS": "asctime, filename, lineno, levelname, message"

Thank you for your help!

**Comment by potiuk** [comment]
Yes. it is expected bycease taks are usually run as separate pods or workers. Your bet is to use GCS or Elasticsearch or similar to store all logs if you want o have them in one place. 

And BTW 1.10.14 is already past end-of-life - it does not receive support not even critical security fixes. Upgrade to Airflow 2 as soon as possible.

**Reply by kangtao** [reply]
Thanks, @potiuk.  To be clear, I believe I was looking at the airflow worker pod logs and I did not see the task run logs. But in scheduler pod, I could see dag_processing logs and scheduler logs. Are you saying this is expected? Is there a way to overwrite task log handler to also write on stdout? Thanks!

**Reply by potiuk** [reply]
I think you can take use as an inspiration  https://airflow.apache.org/docs/apache-airflow/2.1.4/logging-monitoring/logging-tasks.html#advanced-configuration  in Airflow 2. You can likely modify logging settings of airflow as you see fit and send log to output of the process as well - not only to the file.

However 
1) there will be differences in 1.10 that you will have to figure out and since there is no support for 1.10, you might find difficulty in getting someone's help for it
2) depending on what executors your run and how your tasks are running, the output will be sent to different processes output (not scheduler) or they can be swallowed by a process spawning your task processes.

**Reply by potiuk** [reply]
And still your best next move is migrate to Airflow 2 BTW. Any problems with Airflow 1.10, you will find super difficult soon to get any help with.
